oauth2_http/java/com/google/auth/oauth2/UserAuthorizer.java

```diff
@@ -600,7 +638,14 @@ public UserAuthorizer build() {
           transportFactory,
           tokenServerUri,
           userAuthUri,
-          pkce);
+          pkce,
+          clientAuthenticationType);
     }
   }
+
+  public enum ClientAuthenticationType {
```

Comment:
Can you add javadoc here? Also I think this should go at the top but not entirely sure.

Reply:
I spot checked some code consistency in this repo. Seems the public enums goes top and has a javadoc and private enums at the bottom of the class.
I'll move to top and add javadoc.